stdlib/public/core/Zip.swift

```diff
@@ -144,4 +144,15 @@ extension Zip2Sequence: Sequence {
       _sequence1.makeIterator(),
       _sequence2.makeIterator())
   }
+
+  @inlinable // generic-performance
+  public var underestimatedCount: Int {
+    let underestimatedCount1 = _sequence1.underestimatedCount
+    let underestimatedCount2 = _sequence2.underestimatedCount
+
+    return Swift.min(
+      underestimatedCount1 < 0 ? 0 : underestimatedCount1,
+      underestimatedCount2 < 0 ? 0 : underestimatedCount2
```

Comment:
A call to `Swift.max` seems appropriate for these two.

Reply:
Like so?

```swift
  @inlinable // generic-performance
  public var underestimatedCount: Int {
    return Swift.max(0, Swift.min(_sequence1.underestimatedCount, _sequence2.underestimatedCount))
  }
```

Reply:
Yep.

Reply:
Under what circumstances would `underestimatedCount` be a negative value? Do we need the clamp with zero at all?

Reply:
In one example taken from [`MinimalSequence`](https://github.com/apple/swift/blob/c4941fb42716a0bc91b99b7bc36e2cd7b8dd0b1c/stdlib/private/StdlibCollectionUnittest/MinimalCollections.swift#L115) clamping is used to safeguard against negative values when calculating the `underestimatedCount` from some shared mutable state:

```swift
public var underestimatedCount: Int {
  return Swift.max(0, self._sharedState.underestimatedCount - self._sharedState.i)
}
```

Reply:
@dennisvennink Sure, but that's guarding against something that shouldn't ever happen here, right?

Reply:
If the user correctly implements `underestimatedCount` on their own `Sequence` types, then it should never happen. 🙂

Reply:
And if they incorrectly implemented it, they're already in the mess, so the clamp is adding (a tiny bit of) complexity for no benefit. This can just be `min(s1.uC, s2.uC)`.